### DISCUSSION TITLE
Does easeprobe support particular probe ?

### DISCUSSION THREAD

**Discussion by samanhappy** [megaease/easeprobe#172] [Q&A] [answered]
For example, I want to check if a time field is today or not from a json response body, is there any way to do this using easeprobe or google/re2 regex ?

**Comment by haoel** [comment]
@samanhappy with this PR #169, you can do this. it has been merged just now.

**Reply by samanhappy** [reply]
thanks for reply, Uncle Hao, the new TextChecker with RegExp: true is nice, but in some scenarios the require is more specific, so can I write my own TextChecker and associate it with certain probe?

**Reply by haoel** [reply]
You can submit a PR

**Reply by samanhappy** [reply]
I am thinking about introducing some mechanism like plugin to support arbitrary probe, it may take time, do you think it is correct and worthy?

**Reply by haoel** [reply]
May I know the use cases? could you please give me some details on what requirement you want to target? 

Thanks,
Hao

**Reply by samanhappy** [reply]
Let me explain it, We have a http service which return the latest order data, what I want to do is to check if the order time is up to date or not, because if the time is too old, something maybe goes wrong.

I know the requirement is too specific, and I can also acheive it in other ways, so the really question is should or will easeprobe support such specific probes like postman which provide scripts support to do various tests?

BTW, I am trying to achieve this kind of dynamic probe using go plugin, but go plugin has some limits for use, a little bit puzzling...

**Reply by haoel** [reply]
This is really helpful. It seems we need a JSON parse and can do some simple expressions. e.g. now() - order_time < 5m

**Reply by haoel** [reply]
@samanhappy can I have your IM? so that we can have a talk about this feature.

**Reply by samanhappy** [reply]
That will be great, I left a message to you in slack, and submitted a [commit](https://github.com/megaease/easeprobe/commit/281b61deadc539d89604cff592c4ce94eea71799), but not sure to submit a PR, you can review it first.

**Comment by haoel** [comment, marked as answer]
PR #192 has been merged! it should address this issue.
